NullPointerException during integration to Rest API

## This is a...

<pre><code>
[ ] Feature request
[x] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Documentation issue or request
</code></pre>

## Description
I want to create integration AMQ -> Todo app which is described in the [Fuse online tutorials](https://access.redhat.com/documentation/en-us/red_hat_fuse/7.0/html-single/ignite_sample_integration_tutorials/#amq-to-rest-api).
I created AMQ connection, imported extension ([Damage Reporter](https://github.com/syndesisio/fuse-online-sample-extension/releases)), created REST API connector (Customization -> API Client Connectors) and REST API connection for TODO application, created integration.
However, it has never be published and it is stuck in the 'starting' state.

Same situation happens in our test which integrates DB -> Todo custom api connector 

In the integration log in the OpenShift is NPE:
```
2018-10-11 08:54:17.978 ERROR 1 --- [           main] o.s.boot.SpringApplication               : Application startup failed
org.apache.camel.RuntimeCamelException: org.apache.camel.FailedToCreateRouteException: Failed to create route -LOXHluBCImTo_Sgidtn: Route(-LOXHluBCImTo_Sgidtn)[[From[sjms-0-0]] -> [SetHeader[S... because of Failed to create Producer for endpoint: swagger-operation-0-3?operationId=operation-1. Reason: org.apache.camel.RuntimeCamelException: Cannot auto create component: null
	at org.apache.camel.util.ObjectHelper.wrapRuntimeCamelException(ObjectHelper.java:1830) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.spring.SpringCamelContext.start(SpringCamelContext.java:136) ~[camel-spring-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.spring.SpringCamelContext.onApplicationEvent(SpringCamelContext.java:174) ~[camel-spring-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172) ~[spring-context-4.3.17.RELEASE.jar!/:4.3.17.RELEASE]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165) ~[spring-context-4.3.17.RELEASE.jar!/:4.3.17.RELEASE]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139) ~[spring-context-4.3.17.RELEASE.jar!/:4.3.17.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:393) ~[spring-context-4.3.17.RELEASE.jar!/:4.3.17.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:347) ~[spring-context-4.3.17.RELEASE.jar!/:4.3.17.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:883) ~[spring-context-4.3.17.RELEASE.jar!/:4.3.17.RELEASE]
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh(EmbeddedWebApplicationContext.java:144) ~[spring-boot-1.5.13.RELEASE.jar!/:1.5.13.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:546) ~[spring-context-4.3.17.RELEASE.jar!/:4.3.17.RELEASE]
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122) ~[spring-boot-1.5.13.RELEASE.jar!/:1.5.13.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:693) [spring-boot-1.5.13.RELEASE.jar!/:1.5.13.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:360) [spring-boot-1.5.13.RELEASE.jar!/:1.5.13.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:303) [spring-boot-1.5.13.RELEASE.jar!/:1.5.13.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1118) [spring-boot-1.5.13.RELEASE.jar!/:1.5.13.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1107) [spring-boot-1.5.13.RELEASE.jar!/:1.5.13.RELEASE]
	at io.syndesis.example.Application.main(Application.java:13) [classes!/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_151]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_151]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_151]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_151]
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) [project-0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) [project-0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50) [project-0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.PropertiesLauncher.main(PropertiesLauncher.java:595) [project-0.1-SNAPSHOT.jar:na]
Caused by: org.apache.camel.FailedToCreateRouteException: Failed to create route -LOXHluBCImTo_Sgidtn: Route(-LOXHluBCImTo_Sgidtn)[[From[sjms-0-0]] -> [SetHeader[S... because of Failed to create Producer for endpoint: swagger-operation-0-3?operationId=operation-1. Reason: org.apache.camel.RuntimeCamelException: Cannot auto create component: null
	at org.apache.camel.impl.RouteService.warmUp(RouteService.java:147) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.DefaultCamelContext.doWarmUpRoutes(DefaultCamelContext.java:3947) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.DefaultCamelContext.safelyStartRouteServices(DefaultCamelContext.java:3854) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.DefaultCamelContext.doStartOrResumeRoutes(DefaultCamelContext.java:3640) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.DefaultCamelContext.doStartCamel(DefaultCamelContext.java:3492) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.DefaultCamelContext.access$000(DefaultCamelContext.java:209) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.DefaultCamelContext$2.call(DefaultCamelContext.java:3251) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.DefaultCamelContext$2.call(DefaultCamelContext.java:3247) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.DefaultCamelContext.doWithDefinedClassLoader(DefaultCamelContext.java:3270) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.DefaultCamelContext.doStart(DefaultCamelContext.java:3247) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:3163) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.spring.SpringCamelContext.start(SpringCamelContext.java:133) ~[camel-spring-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	... 24 common frames omitted
Caused by: org.apache.camel.FailedToCreateProducerException: Failed to create Producer for endpoint: swagger-operation-0-3?operationId=operation-1. Reason: org.apache.camel.RuntimeCamelException: Cannot auto create component: null
	at org.apache.camel.impl.ProducerCache.doGetProducer(ProducerCache.java:584) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.ProducerCache.acquireProducer(ProducerCache.java:168) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.processor.SendProcessor.doStart(SendProcessor.java:248) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.processor.WrapProcessor.doStart(WrapProcessor.java:52) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.processor.RedeliveryErrorHandler.doStart(RedeliveryErrorHandler.java:1472) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.support.ChildServiceSupport.start(ChildServiceSupport.java:44) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.support.ChildServiceSupport.start(ChildServiceSupport.java:31) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.processor.interceptor.DefaultChannel.doStart(DefaultChannel.java:160) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:62) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.processor.MulticastProcessor.doStart(MulticastProcessor.java:1186) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.processor.WrapProcessor.doStart(WrapProcessor.java:52) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.processor.RedeliveryErrorHandler.doStart(RedeliveryErrorHandler.java:1472) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.support.ChildServiceSupport.start(ChildServiceSupport.java:44) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.support.ChildServiceSupport.start(ChildServiceSupport.java:31) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.processor.interceptor.DefaultChannel.doStart(DefaultChannel.java:160) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:62) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.processor.MulticastProcessor.doStart(MulticastProcessor.java:1186) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.processor.DelegateAsyncProcessor.doStart(DelegateAsyncProcessor.java:80) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.RouteService.startChildService(RouteService.java:370) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.RouteService.doWarmUp(RouteService.java:196) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.RouteService.warmUp(RouteService.java:145) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	... 36 common frames omitted
Caused by: org.apache.camel.RuntimeCamelException: Cannot auto create component: null
	at org.apache.camel.impl.DefaultCamelContext.getComponent(DefaultCamelContext.java:489) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.DefaultCamelContext.getComponent(DefaultCamelContext.java:453) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.spi.RestProducerFactoryHelper.setupComponent(RestProducerFactoryHelper.java:56) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.component.rest.RestEndpoint.createProducer(RestEndpoint.java:350) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.component.rest.swagger.RestSwaggerEndpoint.createProducerFor(RestSwaggerEndpoint.java:286) ~[camel-rest-swagger-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.component.rest.swagger.RestSwaggerEndpoint.createProducer(RestSwaggerEndpoint.java:191) ~[camel-rest-swagger-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.component.connector.DefaultConnectorEndpoint.createProducer(DefaultConnectorEndpoint.java:51) ~[camel-connector-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	at org.apache.camel.impl.ProducerCache.doGetProducer(ProducerCache.java:573) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	... 99 common frames omitted
Caused by: java.lang.NullPointerException: null
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1645) ~[na:1.8.0_151]
	at org.apache.camel.impl.DefaultCamelContext.getComponent(DefaultCamelContext.java:472) ~[camel-core-2.21.0.fuse-720028.jar!/:2.21.0.fuse-720028]
	... 106 common frames omitted

```

Whole log:
[i-amq-12-bhq8t.log](https://github.com/syndesisio/syndesis/files/2468286/i-amq-12-bhq8t.log)

--- a/app/connector/rest-swagger/src/main/resources/camel-connector.json
+++ b/app/connector/rest-swagger/src/main/resources/camel-connector.json
@@ -16,6 +16,7 @@
   "outputDataType" : "json",
   "componentOptions" : [ "host", "basePath" ],
   "endpointValues" : {
+    "componentName" : "http4",
     "consumes" : "application/json"
   },
   "endpointOptions" : [ "operationId" ],

